ODLA/platforms/odla_popart/odla_compute.cc

```diff
@@ -216,7 +217,40 @@ odla_status odla_ExecuteComputation(odla_computation comp, odla_context context,
 
   popart::StepIO stepio(inputs, outputs);
   // Run on ipu
-  g_comp->session->run(stepio);
+  try {
+    g_comp->session->run(stepio);
+  }
+  catch (poplar::application_runtime_error &e) {
+    popart::logging::info("Poplar exception application_runtime_error caught:");
+    return INTERNAL_LOGIC_ERR;
+  }
+  catch (poplar::recoverable_runtime_error &e) {
+    popart::logging::info("Poplar recoverable_runtime_error exception caught");
+    auto action = e.getRecoveryAction();
+    popart::logging::info("need to take action:{}", action);
+    if (action == poplar::RecoveryAction::IPU_RESET) {
+      return RECOVERABLE_ERR;
+    }
+    else if (action == poplar::RecoveryAction::PARTITION_RESET) {
+      return PARTITION_RESET; 
```

Comment:
are these odla_status error code?

Reply:
yes，based on pr#399 https://github.com/alibaba/heterogeneity-aware-lowering-and-optimization/pull/399

Reply:
and need sdk2.2 to use these poplar exceptions.

Reply:
I see. can you change your local branch to ipu_testing_sdk2.2.0 ?